src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = Object.assign({}, state);
+  const out = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can destructure the action to get all the needed props of it (type, extraData, keysToRemove)